p2p/pex/pex_reactor.go

```diff
@@ -475,6 +485,10 @@ func (r *Reactor) ensurePeers() {
 	maxAttempts := numToDial * 3
 
 	for i := 0; i < maxAttempts && len(toDial) < numToDial; i++ {
+		if !r.IsRunning() || !r.book.IsRunning() {
```

Comment:
The first condition here does make much sense, as we don't stop the reactor until this thread is terminates, right?

Reply:
right